osler/inventory/views.py

```diff
@@ -0,0 +1,120 @@
+    template_name = 'inventory/pre_add_new_drug.html'
+    form_class = forms.DuplicateDrugForm
+
+    def form_valid(self, form):
+        name_str = form.cleaned_data['name'].capitalize()
+        lot_number_str = form.cleaned_data['lot_number'].upper()
+        manufacturer_str = form.cleaned_data['manufacturer']
+
+        querystr = '%s=%s&%s=%s&%s=%s' % ("name", name_str,
+                                    "lot_number", lot_number_str,
+                                    "manufacturer", manufacturer_str)
+
+        add_new_drug_url = "%s?%s" % (reverse("inventory:drug-add-new"), querystr)
+
+        if lot_number_str.strip() == '':
+            return HttpResponseRedirect(add_new_drug_url)
+
+        matching_drugs = models.Drug.objects.filter(name=name_str, lot_number=lot_number_str, manufacturer=manufacturer_str)
+
+        if len(matching_drugs) > 0:
+            predrug_select_url = "%s?%s" % (reverse("inventory:predrug-select"), querystr)
+            return HttpResponseRedirect(predrug_select_url)
+
+        return HttpResponseRedirect(add_new_drug_url)
+
+class PreDrugSelect(ListView):
+    template_name = 'inventory/predrug-select.html'
+    new_drug_url = ""
+
+    def parse_url_querystring(self):
+
+        return utils.get_name_and_lot_from_url_query_dict(self.request)
+
+    def get_queryset(self):
+        initial = self.parse_url_querystring()
+        if (initial.get('name', None) is None):
+            return []
+        possible_duplicates = models.Drug.objects.filter(name=initial.get('name', None),\
+                                                        lot_number=initial.get('lot_number', None),\
+                                                        manufacturer=initial.get('manufacturer', None))
+        return possible_duplicates
+
+    def get_context_data(self, **kwargs):
+        # Call the base implementation first to get a context
+        context = super(PreDrugSelect, self).get_context_data(**kwargs)
+        initial = self.parse_url_querystring()
+        context['name'] = initial.get('name', None)
+        context['lot_number'] = initial.get('lot_number', None)
+        context['manufacturer'] = initial.get('manufacturer', None)
+        context['new_drug_url'] = "%s?%s=%s&%s=%s&%s=%s" % (
+            reverse("inventory:drug-add-new"),
+            "name", initial.get('name', None),
+            "lot_number", initial.get('lot_number', None),
+            "manufacturer", initial.get('manufacturer', None))
+        context['home'] = reverse("inventory:drug-list")
+        return context
+
+class DrugAddNew(FormView):
+    template_name = 'inventory/add_new_drug.html'
+    form_class = forms.DrugForm
+
+    def form_valid(self, form):
+        df = form.save()
+        df.save()
+        return redirect('inventory:drug-list')
+
+    def get_initial(self):
+        initial = super(DrugAddNew, self).get_initial()
+
+        initial.update(utils.get_name_and_lot_from_url_query_dict(self.request))
```

Comment:
What is the purpose of this? You don't pass a specific drug into add-new do you?

Reply:
The purpose of this is that there is a PreDrugAddNew class where the user initially enters the drug name, lot number, and manufacturer. Then if there is not drug with the same attributes, the user is led to the DrugAddNew class. The get_initial function autopopulates the fields with what the user already entered in PreDrugAddNew.